Symfony/CS/Fixer/Symfony/NoEmptyStatementFixer.php

```diff
@@ -0,0 +1,189 @@
+
+        if ($tokens[$beforeCurlyOpening]->isGivenKind($beforeCurlyOpeningKinds) || $tokens[$beforeCurlyOpening]->equals(';')) {
+            $tokens->clearTokenAndMergeSurroundingWhitespace($index);
+
+            return;
+        }
+
+        // check for class, interface and trait definitions
+        if ($tokens[$beforeCurlyOpening]->isGivenKind(T_STRING) && $this->isNonAnonymousClassClassyDefinition($tokens, $tokens->getPrevMeaningfulToken($beforeCurlyOpening))) {
+            $tokens->clearTokenAndMergeSurroundingWhitespace($index);
+
+            return;
+        }
+
+        // early return check, below only control structures with conditions are fixed
+        if (!$tokens[$beforeCurlyOpening]->equals(')')) {
+            return;
+        }
+
+        $openingBrace = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $beforeCurlyOpening, false);
+        $beforeOpeningBrace = $tokens->getPrevMeaningfulToken($openingBrace);
+
+        if ($tokens[$beforeOpeningBrace]->isGivenKind(array(T_IF, T_ELSEIF, T_FOR, T_FOREACH, T_WHILE, T_SWITCH, T_CATCH, T_DECLARE))) {
+            $tokens->clearTokenAndMergeSurroundingWhitespace($index);
+
+            return;
+        }
+
+        // check for function definition
+        if ($tokens[$beforeOpeningBrace]->isGivenKind(T_STRING)) {
+            $beforeString = $tokens->getPrevMeaningfulToken($beforeOpeningBrace);
+            if (null !== $beforeString && $tokens[$beforeString]->isGivenKind(T_FUNCTION)) {
+                $tokens->clearTokenAndMergeSurroundingWhitespace($index); // implicit return
+            }
+        }
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return bool
+     */
+    private function isAnonymousClass(Tokens $tokens, $index)
+    {
+        return $tokens[$tokens->getPrevMeaningfulToken($index)]->isGivenKind(T_NEW);
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $classyTest
+     *
+     * @return bool
+     */
+    private function isNonAnonymousClassClassyDefinition(Tokens $tokens, $classyTest)
+    {
+        // check for class definition
+        if ($tokens[$classyTest]->isGivenKind(T_CLASS)) {
+            return !$this->isAnonymousClass($tokens, $classyTest);
+        }
+
+        // check for interface or trait definition (note: class definition is already check above)
+        if ($tokens[$classyTest]->isClassy()) {
+            return true;
+        }
+
+        do {
+            $classyTest = $tokens->getPrevMeaningfulToken($classyTest);
+        } while ($tokens[$classyTest]->equals(',') || $tokens[$classyTest]->isGivenKind(array(T_STRING, T_EXTENDS, T_IMPLEMENTS)));
+
+        if ($tokens[$classyTest]->equals(')')) {
+            return false;
+        }
+
+        if (!$tokens[$classyTest]->isGivenKind(T_CLASS)) {
+            return true;
+        }
+
+        return !$this->isAnonymousClass($tokens, $classyTest);
```

Comment:
somehow that will break this test atm;
https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1671/files#diff-1c0ce34bfa55fca688255efb4c1025dcR431

I'll look into it. Thanks for reviewing and all the pointers!